src/types/table.ts

```diff
@@ -13,3 +13,7 @@ export enum Order {
   ASC = 'asc',
   DESC = 'desc',
 }
+
+export type AppActionData = ActionData & {
+  [key: string]: string; // Extending ActionData to include the index signature
+};
```

Comment:
With this you effectively loose typing on the data property. 
Try for example to use `rows[0].data.toto` and it will tell you that it is a string even when `toto` does not exist on the `ActionData` type.

I am guessing that the issue you have is that the `ActionData` interface does not satisfy the generic `Data` constraint on `appAtion.data`. 

This is the TS error:
```
Type 'ActionData' does not satisfy the constraint 'Data'.
  Index signature for type 'string' is missing in type 'ActionData'.
```
You can fix this by replacing the interface on line 7 by a type alias.
And you do not need to extend it with the index signature. 
You can then use `ActionData` from line 7 directly for the generic type parameter on `AppAction`.